crates/status/src/status_manager.rs

```diff
@@ -16,93 +20,147 @@ pub enum StatusError {
+        self.receiver
+            .chs
+            .borrow()
+            .clone()
+            .map(|chs| chs.operator_table().clone())
     }
 
-    pub fn l1(&self) -> &watch::Receiver<L1Status> {
-        &self.l1
+    /// Gets the latest deposits table.
+    pub fn deposits_table(&self) -> Option<DepositsTable> {
+        self.receiver
+            .chs
+            .borrow()
+            .clone()
+            .map(|chs| chs.deposits_table().clone())
     }
 
-    pub fn chs(&self) -> &watch::Receiver<Option<Chainstate>> {
-        &self.chs
+    /// Gets the latest [`L1Status`].
+    pub fn l1_status(&self) -> L1Status {
+        self.receiver.l1.borrow().clone()
+    }
+
+    /// Waits until there's a new client state and returns the client state.
+    pub async fn wait_for_client_change(&self) -> Result<ClientState, RecvError> {
+        let mut s = self.receiver.cl.clone();
+        s.changed().await?;
+        let state = s.borrow().clone();
+        Ok(state)
     }
-}
 
-impl StatusTx {
-    pub fn update_chain_state(&self, post_state: &Chainstate) {
-        if self.chs.send(Some(post_state.clone())).is_err() {
+    /// Waits until genesis and returns the client state.
+    pub async fn wait_until_genesis(&self) -> Result<ClientState, RecvError> {
+        let mut rx = self.receiver.cl.clone();
+        loop {
+            if rx.borrow().has_genesis_occurred() {
+                return Ok(rx.borrow().clone());
+            }
+            rx.changed().await?;
+        }
+    }
+
+    // Sender methods
+
+    /// Sends the updated `Chainstate` to the chain state receiver. Logs a warning if the receiver
+    /// is dropped.
+    pub fn update_chainstate(&self, post_state: &Chainstate) {
+        if self.sender.chs.send(Some(post_state.clone())).is_err() {
             warn!("chain state receiver dropped");
         }
     }
 
+    /// Sends the updated `ClientState` to the client state receiver. Logs a warning if the receiver
+    /// is dropped.
     pub fn update_client_state(&self, post_state: &ClientState) {
-        if self.cl.send(post_state.clone()).is_err() {
+        if self.sender.cl.send(post_state.clone()).is_err() {
             warn!("client state receiver dropped");
         }
     }
 
+    /// Sends the updated `L1Status` to the L1 status receiver. Logs a warning if the receiver is
+    /// dropped.
     pub fn update_l1_status(&self, post_state: &L1Status) {
-        if self.l1.send(post_state.clone()).is_err() {
+        if self.sender.l1.send(post_state.clone()).is_err() {
             warn!("l1 status receiver dropped");
         }
     }
 
-    pub fn update_csm_status(&self, post_state: &CsmStatus) {
-        if self.csm.send(post_state.clone()).is_err() {
-            warn!("csm status receiver dropped");
-        }
+    /// Subscribes to updates for `ClientState` and returns a receiver.
+    pub fn subscribe_cl(&self) -> watch::Receiver<ClientState> {
+        self.sender.cl.subscribe()
```

Comment:
This doesn't seem to be used anywhere, ideally we shouldn't leak it.

Reply:
Oh yes, I forgot it. Removed.